### ISSUE TITLE
hdf5widget: add import hdf5plugin before h5py

### ISSUE THREAD

**Issue by vincefn** [silx-kit/silx#571]
try importing hdf5plugin before h5py, so that some compressed files can be opened without being struck with the "can't open directory" error.
Maybe advise to add hdf5plugins if a dataset cannot be read with an OSError

**Comment by vasole** [comment]
It is more a responsibility of the application. We can put it in the examples.

**Comment by vincefn** [comment]
Yes, I was thinking about adding an import to the hdf5widget.py in silx/examples:

try:
    # This import must be done before h5py
    import hdf5plugin
except:
    print("You should install the hdf5plugin package [url] to support XX hdf5 compression")

**Comment by vallsv** [comment]
Yes, it is exactly what it will be done

**Comment by vasole** [comment]
It seems HDF5 or h5py do not properly set the dataset attribute compression when an external filter has been used for compression.
 
At least I could found a way to know what filters have been used on a dataset:

```
import h5py
h = h5py.File("lz4.h5", "r")
d = h["/entry/data"]
for i in range(d._dcpl.get_num_filters()):
    print(d._dcpl.get_filter(i)
```

**Comment by vincefn** [comment]
That would be very useful. Only catch, is that if they follow PEP, _dcpl is a non-public API. Would be nice to have a stable access.

**Comment by vasole** [comment]
Ok, let's be purists although I do not think they will change that.

I have taken a look at how _dcpl is generated and we can replace it by the public API

```
import h5py
h = h5py.File("lz4.h5", "r")
d = h["/entry/data"]
dcpl = d.id.get_create_plist()
for i in range(dcpl.get_nfilters()):
    print(dcpl.get_filter(i))
```
